### DISCUSSION TITLE
Help with WireGuard Android source

### DISCUSSION THREAD

**Discussion by xcpn** [ImranR98/Obtainium#1577] [Q&A]
I've been trying to add the WireGuard Android source link for half an hour, and every variation I try I get errors.

Here is some info:

- Source link: https://download.wireguard.com/android-client/

- Custom APK link filter: com.wireguard.android-*.apk

- Filter links by link text [ON]

- Track-Only [ON]

Trying to add this, results in error "**Could not find a suitable version**".

Trying to change Override Source from 'None' to 'HTML' results in error "**Could not find a suitable version**".

Trying to change Override Source from 'None' to 'Direct APK Link' results in error "**Internal Error**".

Trying to add the Source as is without changing anything, results in no error and it's added. Upon trying to download asset via 'Download release asset', results in error "**PathNotFoundException: Cannot open file, path = '/storage/emulated/0/Download/https://download.wireguard.com/android-client/com.wireguard.android-1.0.20231018.apk.part' (OS Error: No such file or directory, errno = 2) [https://download.wireguard.com/android-client/com.wireguard.android-1.0.20231018.apk]**"

Is this possible or I'm trying for nothing, any help appreciated!

**Comment by ImranR98** [comment]
You shouldn't need to change any settings - pasting the URL you provided and clicking 'Add' is enough (if you want APK extraction from the link, `wireguard\.android-(.+)\.apk` with match group 1 works).

You might get an "Internal Server Error" but it will work if you retry a few times. The error itself comes from the webpage, not Obtainium. I got the same error a couple of times when visiting the page in a browser - it seems the site is just unreliable.

**Reply by xcpn** [reply]
Where should I add _wireguard\.android-(.+)\.apk_ so I can download the APK?

Adding it to Custom APK Link Filter results in the error I mentioned before when trying to download asset.

**Reply by ImranR98** [reply]
In the 'APK extract' field. It's not strictly needed, it just lets you get a more accurate version string. I recommend reading the wiki.

**Reply by xcpn** [reply]
The versioning is correct, but I can't download the APK via the _Download release asset_ however.

**Reply by ImranR98** [reply]
Okay I see the issue. Can you open a bug report?

**Reply by xcpn** [reply]
Will do, let me try right now!

**Reply by xcpn** [reply]
@ImranR98 tried my best.

https://github.com/ImranR98/Obtainium/issues/1578
